### PR TITLE
Add dashboard widget

### PR DESCRIPTION
**Pull request by berdosi**
This request implements a very basic Weather widget for the Nextcloud Dashboard: it shows the most important weather data for the currently selected home city.

screenshot-190811-233801

### CODE REVIEW

**Comment by nerzhul** [comment]
Great job. Is this ready for a merge ?

**Comment by berdosi** [comment]
Now that you mention it, right, it isn't:
* The widget doesn't report errors. 
* There's a leftover console.info() in the JavaScript part; I'll lint it through for more findings.

I'll comment again once these are sorted out, probably in the coming days.

**Comment by berdosi** [comment]
Hello again! I consider it ready for a merge now.

File: lib/Widgets/DefaultWidget.php [hunk 1]
```diff
@@ -111,32 +114,36 @@ public function loadWidget(IWidgetConfig $settings) {
...
-			$result = json_decode($weatherController->get($firstCity)->render(), true);
-			$metric = json_decode($settingsController->metricGet()->render(), true)['metric'];
-
-			$request->addResult('location', $firstCity);
-			$request->addResult('temperature', $result['main']['temp']);
-			$request->addResult('metric', $metric);
-			$request->addResult('weather', $result['weather'][0]['description']);
-			$request->addResult('humidity', $result['main']['humidity']);
-			$request->addResult('wind', $result['wind']['speed']);
+			try {
```
Comment (nerzhul) [inline lib/Widgets/DefaultWidget.php]: please don't catch on all the block, only required block

Comment (berdosi) [reply lib/Widgets/DefaultWidget.php]: Quite right. 
As there were only two points of failure they could be checked separately – thus there isn't even need for a try/catch.

**Comment by e-alfred** [comment]
@nerzhul @berdosi Now that all issues are fixed, do you think we can merge this request into the main repository?

**Comment by nerzhul** [comment]
@e-alfred can you verify it's valid against current master and test it ? code looks good to me currently

**Comment by e-alfred** [comment]
@berdosi I think the error handling could be improved for a better user experience:

- Show an explicit message that no city is defined in the Weather app
- Show an explicit message if no home city is defined in the Weather app

Maybe in a future improvement additional cities/towns can be added than just the home city.

**Comment by e-alfred** [comment]
@nerzhul It works well with the latest master branch, I think we could merge it.

**Comment by e-alfred** [comment]
@nerzhul I published a new release on the release page, please publish it to the Nextcloud App Store as soon as possible.

**Comment by nerzhul** [comment]
i'll do it

**Comment by nerzhul** [comment]
@e-alfred can you tag as 1.7.0, it's a new feature not a patch, thanks :)

**Comment by e-alfred** [comment]
I uploaded a new release as 0.1.7, please publish it to the App Store.

**Comment by e-alfred** [comment]
@berdosi It seems like the old Dashboard was deprecated and a new one is implemented into the core: https://github.com/nextcloud/server/issues/20930

**Comment by berdosi** [comment]
Thanks for the heads up!
I had a  quick look at the PR implementing the widget in Talk ( https://github.com/nextcloud/spreed/pull/3890 ), and it doesn't look like we can share a lot of code (maybe except for the widget's template, that ain't a lot).

**Comment by nerzhul** [comment]
it's nice if we can be up to date, nice job :)

**Comment by e-alfred** [comment]
Nextcloud server now has a weather dashboard app created by @eneiluj. Maybe we can collaborate on integrating both apps better in the future?

**Comment by berdosi** [comment]
Indeed, I followed this PR quite closely, maybe not close enough, though:) →  https://github.com/nextcloud/server/pull/22124
- from widget's point of view, the one-line status shows pretty much the same info as current (#76) did, but in a sleeker way.
- currently clicking on the new widget brings us to the weather service  - it may be good to be able to hook into that :)
- reusing the same backend would be good as two weather services within Nextcloud may be one too many :) Currently that one is polling https://api.met.no/weatherapi/locationforecast/2.0/documentation#!/data/get_compact which doesn't show the same detail that the Weather app does – the "complete" endpoint seems to be better for our purposes.

**Pull request merged**